### PR TITLE
[release/v0.1] Fix for panic and to allow GRB deletion

### PR DESCRIPTION
**Pull request by ryansann**
Issue:
- https://github.com/rancher/rancher/issues/33429

These changes pull in a new wrangler version which has a fix for a panic in the webhook handler logic. Also included is allowance for update & delete of GRBs if the corresponding GR isn't found, i.e. it was deleted. Updates to GRBs are only allowed if the DeletionTimestamp != nil, this only allows updates to finalizers when there is no associated GR

### CODE REVIEW

File: pkg/resources/globalrolebinding/globarolebinding.go [hunk 1]
```diff
@@ -36,6 +40,27 @@ func (grbv *globalRoleBindingValidator) Admit(response *webhook.Response, reques
 	// Pull the global role to get the rules
 	globalRole, err := grbv.globalRoles.Get(newGRB.GlobalRoleName)
 	if err != nil {
+		if errors.IsNotFound(err) {
```
Comment (dramich) [inline pkg/resources/globalrolebinding/globarolebinding.go:43]: Can we do an opposite check here and then return if it isn't then the fallthrough is what exists inside this if check

File: pkg/resources/globalrolebinding/globarolebinding.go [hunk 2]
```diff
@@ -36,6 +40,27 @@ func (grbv *globalRoleBindingValidator) Admit(response *webhook.Response, reques
...
+				return nil
+			// only allow updates to the finalizers if the GR associated to the GRB is not found
+			case admissionv1.Update:
+				if newGRB.DeletionTimestamp != nil {
+					response.Allowed = true
+					return nil
+				}
+				response.Result = &metav1.Status{
+					Status:  "Failure",
+					Message: fmt.Sprintf("GlobalRoleBinding %s associated GlobalRole not found. Only deletion and updates to finalizers are permitted.", newGRB.Name),
```
Comment (dramich) [inline pkg/resources/globalrolebinding/globarolebinding.go:57]: Errors should start with lowercase and this message is confusing. Can it be updated to something like `referenced globalRole %v not found, updates not allowed`. I don't think we need to call out the finalizer stuff as that's an implementation detail of rancher and how we object lifecycles and probably won't be useful to a user.

Comment (ryansann) [reply pkg/resources/globalrolebinding/globarolebinding.go:57]: Yes, agreed, I was trying to figure out a good way to say this. I'll change it to this

File: pkg/resources/globalrolebinding/globarolebinding.go (hunk 1 above)
Comment (ryansann) [reply pkg/resources/globalrolebinding/globarolebinding.go:43]: Yep, I'll change

File: pkg/resources/globalrolebinding/globarolebinding.go [hunk 3]
```diff
@@ -36,7 +40,28 @@ func (grbv *globalRoleBindingValidator) Admit(response *webhook.Response, reques
...
+		case admissionv1.Delete: // allow delete operations if the GR is not found
+			response.Allowed = true
+			return nil
+		case admissionv1.Update: // only allow updates to the finalizers if the GR is not found
+			if newGRB.DeletionTimestamp != nil {
+				response.Allowed = true
+				return nil
+			}
+			fallthrough
+		default: // other operations not allowed when GR is not found
```
Comment (dramich) [inline pkg/resources/globalrolebinding/globarolebinding.go:56]: These lines aren't needed since default and no match is the same operation

**Review by dramich — APPROVED** [review]

**Pull request merged**